packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1801,22 +1799,26 @@ export class ContainerRuntime
 		const snapshotSeqNumber = snapshot.sequenceNumber;
 		assert(snapshotSeqNumber !== undefined, "snapshotSeqNumber should be present");
 
+		// This assert fires if we get a snapshot older than the snapshot we loaded from. This is a service issue.
+		assert(
+			snapshotSeqNumber >= this.deltaManager.initialSequenceNumber,
+			"Downloaded snapshot older than snapshot we loaded from",
+		);
+
 		// If the snapshot is ahead of the last seq number of the delta manager, then catch up before
 		// returning the snapshot.
 		if (snapshotSeqNumber > this.deltaManager.lastSequenceNumber) {
 			// If this is a summarizer client, which is trying to load a group and it finds that there is
 			// another snapshot from which the summarizer loaded and it is behind, then just give up as
 			// the summarizer state is not up to date.
+			// This should be a recoverable scenario and shouldn't happen as we should process the ack first.
 			if (this.isSummarizerClient) {
 				throw new Error(
-					"Summarizer client behind when loading snapshot with loadingGroupId",
+					"Summarizer client behind, loaded newer snapshot with loadingGroupId",
 				);
 			}
-			// If the inbound deltas queue is paused, then unpause it for now as we need
-			// to catch up to the snapshot sequence number.
-			if (this.deltaManager.inbound.paused) {
```

Comment:
worth checking, but I think pausing should not happen with op grouping On (it's not enabled in main, but enabled 100% in Loop). Same for initial load (paused) and delay-loading (delay-loading should happen after unpause).
If that's the case, I'd consider adding a telemetry event here if queue is paused. It could be very valuable in debugging if something bad happens.

Overall Q: Do we have telemetry to tell us things like
- how often delay loading is used
- how long it takes to load?
To some extent, both are covered in driver, so maybe we are fine here, but worth considering - what kind of telemetry we want for
- general understanding of how system is used
- debugging.

Reply:
Yep, just improved the telemetry to do so. We have `GroupIdSnapshotFetched` to see all the `groupId` snapshot fetch events, the driver events for how long it takes to retreive from storage, and `GroupIdSnapshotCatchup` to see all the events where catchup is required.